### PR TITLE
Update to PRT 2.2 and basic array attribute support

### PR DESCRIPTION
**Pull request by mistafunk**
This is a big PR covering multiple topics:
* Update to PRT 2.2 (includes requiring MSVC 14.2 on Windows)
* Adding array attribute support both when creating initial shapes and when re-emitting them onto the generated models.
* Fixing the unit tests.
* Pruning Houdini 17.0 from the CI pipeline.
* Various code cleanups and small improvements.

### CODE REVIEW

File: src/palladio/ShapeGenerator.cpp [hunk 1]
```diff
@@ -111,35 +115,104 @@ void ShapeGenerator::get(const GU_Detail* detail, const PrimitiveClassifier& pri
...
+						}
+					}
+					else {
+						GA_ROHandleD av(ar);
+						if (av.isValid()) {
+							double v = av.get(primitiveMapOffset);
+							if (DBG)
+								LOG_DBG << "   prim float attr: " << ar->getName() << " = " << v;
+							amb->setFloat(ruleAttrName.c_str(), v);
+						}
```
Comment (bneukom) [inline src/palladio/ShapeGenerator.cpp:135]: Code here (and in other cases) is nested quite deeply. Could maybe be moved into functions to improve readability a bit?

File: src/palladio/AttributeConversion.cpp [hunk 2]
```diff
@@ -91,6 +91,52 @@ void setHandleRange(const GA_IndexMap& indexMap, GA_RWBatchHandleS& handle, GA_O
...
+
+void setHandleRange(const GA_IndexMap& indexMap, GA_RWHandleIA& handle, GA_Offset start, GA_Size size, const bool* ptr,
+                    size_t ptrSize) {
+	UT_IntArray hv(ptrSize, ptrSize); // there is no UT_Int8Array
+	for (size_t i = 0; i < ptrSize; i++)
+		hv[i] = ptr[i] ? 1u : 0u;
+	for (GA_Offset off = start; off < start + size; off++)
+		handle.set(off, hv);
+	if (DBG)
+		LOG_DBG << "int array attr: range = [" << start << ", " << start + size
```
Comment (bneukom) [inline src/palladio/AttributeConversion.cpp:116]:
```suggestion
		LOG_DBG << "bool array attr: range = [" << start << ", " << start + size
```

File: src/palladio/AttributeConversion.cpp [hunk 3]
```diff
@@ -91,6 +91,52 @@ void setHandleRange(const GA_IndexMap& indexMap, GA_RWBatchHandleS& handle, GA_O
...
+
+void setHandleRange(const GA_IndexMap& indexMap, GA_RWHandleSA& handle, GA_Offset start, GA_Size size,
+                    wchar_t const* const* ptr, size_t ptrSize) {
+	UT_StringArray hv(ptrSize, ptrSize);
+	for (size_t i = 0; i < ptrSize; i++)
+		hv[i] = UT_StringHolder(toOSNarrowFromUTF16(ptr[i]));
+	for (GA_Offset off = start; off < start + size; off++)
+		handle.set(off, hv);
+	if (DBG)
+		LOG_DBG << "int array attr: range = [" << start << ", " << start + size
```
Comment (bneukom) [inline src/palladio/AttributeConversion.cpp:128]:
```suggestion
		LOG_DBG << "string array attr: range = [" << start << ", " << start + size
```

File: src/palladio/ShapeGenerator.cpp (hunk 1 above)
Comment (mistafunk) [reply src/palladio/ShapeGenerator.cpp]: Factored out the conversion

**Comment by mistafunk** [comment]
Thanks for review. I've factored out the attribute conversion and made it a bit more symmetrical. Still some more opportunities to deduplicate the code in AttributeConversion.cpp but I'll stop here for now.

File: src/palladio/AttributeConversion.h [hunk 4]
```diff
@@ -39,6 +40,151 @@ struct hash<UT_String> {
 
 namespace AttributeConversion {
 
+class FromHoudini {
+private:
+	static const bool DBG = true;
```
[the remark this answers is not in the document]
Comment (mistafunk) [reply src/palladio/AttributeConversion.h]: that's already fixed with the force push above

**Pull request merged**